src/solvers/dg/dg_1d.jl

```diff
@@ -0,0 +1,465 @@
+  @unpack surface_flux_values = cache.elements
+  @unpack n_boundaries_per_direction = cache.boundaries
+
+  # Calculate indices
+  lasts = accumulate(+, n_boundaries_per_direction)
+  firsts = lasts - n_boundaries_per_direction .+ 1
+
+  # Calc boundary fluxes in each direction
+  calc_boundary_flux_by_direction!(surface_flux_values, t, boundary_conditions[1],
+                                   equations, dg, cache, 1, firsts[1], lasts[1])
+  calc_boundary_flux_by_direction!(surface_flux_values, t, boundary_conditions[2],
+                                   equations, dg, cache, 2, firsts[2], lasts[2])
+end
+
+function calc_boundary_flux_by_direction!(surface_flux_values::AbstractArray{<:Any,3}, t,
+                                          boundary_condition, equations, dg::DG, cache,
+                                          direction, first_boundary, last_boundary)
+  @unpack surface_flux = dg
+  @unpack u, neighbor_ids, neighbor_sides, node_coordinates, orientations = cache.boundaries
+
+  Threads.@threads for boundary in first_boundary:last_boundary
+    # Get neighboring element
+    neighbor = neighbor_ids[boundary]
+
+    # Get boundary flux
+    u_ll, u_rr = get_surface_node_vars(u, equations, dg, boundary)
+    if neighbor_sides[boundary] == 1 # Element is on the left, boundary on the right
+      u_inner = u_ll
+    else # Element is on the right, boundary on the left
+      u_inner = u_rr
+    end
+    x = get_node_coords(node_coordinates, equations, dg, boundary)
+    flux = boundary_condition(u_inner, orientations[boundary], direction, x, t, surface_flux,
+                              equations)
+
+    # Copy flux to left and right element storage
+    for v in eachvariable(equations)
+      surface_flux_values[v, direction, neighbor] = flux[v]
+    end
+  end
+
+  return nothing
+end
+
+
+function calc_surface_integral!(du::AbstractArray{<:Any,3}, equations, dg::DGSEM, cache)
+  @unpack boundary_interpolation = dg.basis
+  @unpack surface_flux_values = cache.elements
+
+  Threads.@threads for element in eachelement(dg, cache)
+    for v in eachvariable(equations)
+      # surface at -x
+      du[v, 1,          element] -= surface_flux_values[v, 1, element] * boundary_interpolation[1,          1]
+      # surface at +x
+      du[v, nnodes(dg), element] += surface_flux_values[v, 2, element] * boundary_interpolation[nnodes(dg), 2]
+    end
+  end
+
+  return nothing
+end
+
+
+function apply_jacobian!(du::AbstractArray{<:Any,3}, equations, dg::DG, cache)
+
+  Threads.@threads for element in eachelement(dg, cache)
+    factor = -cache.elements.inverse_jacobian[element]
+
+    for i in eachnode(dg)
+      for v in eachvariable(equations)
+        du[v, i, element] *= factor
+      end
+    end
+  end
+
+  return nothing
+end
+
+
+# TODO: Taal dimension agnostic
+# TODO: Taal refactor, make source terms pointwise, possibly using traits to decide if they depend on u, if necessary for performance?
```

Comment:
```suggestion
```
Tracked in #208

Reply:
Thanks for creating the issue. I would still like to keep the notes there for now - a bit of redundancy is often useful, I think.